src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -211,304 +219,28 @@ private static void EscapeNextBytes(byte value, Span<byte> destination, ref int
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static bool IsAsciiValue(byte value) => value <= LastAsciiCharacter;
 
-        private static bool IsAsciiValue(char value) => value <= LastAsciiCharacter;
-
-        /// <summary>
-        /// Returns <see langword="true"/> if <paramref name="value"/> is a UTF-8 continuation byte.
-        /// A UTF-8 continuation byte is a byte whose value is in the range 0x80-0xBF, inclusive.
-        /// </summary>
-        private static bool IsUtf8ContinuationByte(byte value) => (value & 0xC0) == 0x80;
-
-        /// <summary>
-        /// Returns <see langword="true"/> if the low word of <paramref name="char"/> is a UTF-16 surrogate.
-        /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static bool IsLowWordSurrogate(uint @char)
-            => (@char & 0xF800U) == 0xD800U;
-
-        // We can't use the type Rune since it is not available on netstandard2.0
-        // To avoid extensive ifdefs and for simplicity, just using an int to reprepsent the scalar value, instead.
-        public static SequenceValidity PeekFirstSequence(ReadOnlySpan<byte> data, out int numBytesConsumed, out int rune)
-        {
-            // This method is implemented to match the behavior of System.Text.Encoding.UTF8 in terms of
-            // how many bytes it consumes when reporting invalid sequences. The behavior is as follows:
-            //
-            // - Some bytes are *always* invalid (ranges [ C0..C1 ] and [ F5..FF ]), and when these
-            //   are encountered it's an invalid sequence of length 1.
-            //
-            // - Multi-byte sequences which are overlong are reported as an invalid sequence of length 2,
-            //   since per the Unicode Standard Table 3-7 it's always possible to tell these by the second byte.
-            //   Exception: Sequences which begin with [ C0..C1 ] are covered by the above case, thus length 1.
-            //
-            // - Multi-byte sequences which are improperly terminated (no continuation byte when one is
-            //   expected) are reported as invalid sequences up to and including the last seen continuation byte.
-
-            Debug.Assert(JsonHelpers.IsValidUnicodeScalar(ReplacementChar));
```

Comment:
Can we get rid of some of the other helpers that are no longer needed, like this one `JsonHelpers.IsValidUnicodeScalar`?

Reply:
I'll check. I believe IsValidUnicodeScalar is used by NetFx, but could ifdef that then